lib/department_api.rb

```diff
@@ -6,40 +6,35 @@ def initialize(access_token)
   end
 
   def get_departments_info(dept_name)
-    url = URI("https://gw.api.it.umich.edu/um/bf/Department/DeptData?deptDescr=#{dept_name}")
+    begin 
+      url = URI("https://gw.api.it.umich.edu/um/bf/Department/v2/DeptData?DeptDescription=#{dept_name}")
 
-    http = Net::HTTP.new(url.host, url.port)
-    http.use_ssl = true
-    http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+      http = Net::HTTP.new(url.host, url.port)
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_NONE
 
-    request = Net::HTTP::Get.new(url)
-    request["x-ibm-client-id"] = "#{Rails.application.credentials.um_api[:buildings_client_id]}"
-    request["authorization"] = "Bearer #{@access_token}"
-    request["accept"] = 'application/json'
+      request = Net::HTTP::Get.new(url)
+      request["x-ibm-client-id"] = "#{Rails.application.credentials.um_api[:buildings_client_id]}"
+      request["authorization"] = "Bearer #{@access_token}"
+      request["accept"] = 'application/json'
 
-    response = http.request(request)
-    response_json = JSON.parse(response.read_body)
-    # if dept_name == "EH&S"
-    #   puts response_json
-    # end
-
-    if response_json.present?
-      if response_json['errorCode'].present?
-        @result['errorcode'] = response_json['errorCode']
-        @result['error'] = response_json['errorMessage']
-      elsif response_json['ErrorResponse'].present?
-        @result['errorcode'] = response_json['ErrorResponse']['responseCode'].to_s
-        @result['error'] = response_json['ErrorResponse']['responseDescription']
-      elsif response_json['DepartmentList'].present?
+      response = http.request(request)
+      response_json = JSON.parse(response.read_body)
+      if response.code == "200"
         @result['success'] = true
         @result['data'] = response_json['DepartmentList']
       else
-        @result['error'] = 'Unknown error'
+        if response_json['errorCode'].present?
+          @result['errorcode'] = response_json['errorCode']
+          @result['error'] = response_json['errorMessage']
+        else 
+          @result['errorcode'] = "Unknown error"
```

Comment:
I can fake an 'Invalid access token' error:
`{"errorCode"=>"ERR401", "errorMessage"=>"Invalid access token"}`
but, otherwise, at the moment there are no other errors.

Reply:
What is the value of response.code in this case ?

Reply:
Oh, sorry, did not see the word code in your question. 
For the  invalid access token error the response code is 401.

Reply:
Thanks.